### ISSUE TITLE
BelongsTo and HasMany not supported in nested forms

### ISSUE THREAD

**Issue by Billiam** [graysonwright/administrate-field-nested_has_many#5]
I believe that these two field types (and possibly select as well), will not work correctly within nested-has-many fields.

Unlike other fields, which don't bother explicitly passing field values (using the formbuilder instead), these types rely directly on field `data` to populate selected options: 

Example from belongs_to/_form.html.erb
```erb
<div class="field-unit__field">
  <%= f.select(field.permitted_attribute) do %>
    <%= options_for_select(field.associated_resource_options, field.selected_option) %>
  <% end %>
</div>
```

```ruby
def selected_option
  data && data.id
end
```

I don't think this is an issue with the nested has many field type, but does show that upstream fields may not play nicely with it.

It seems like it would be safe to override these fields' templates, ex:

```erb
<%= f.select(field.permitted_attribute, field.associated_resource_options) %>
```

but I hesitate to issue a pull request to administrate without knowing the reason these fields are handled differently, or whether the nested has many field represents a supported use case.

**Comment by michalvalasek** [comment]
TL;DR: the solution proposed by @Billiam works.

My use case:
I had this exact problem (**Recipe** has_many_nested **Tools** belongs_to **Product**). When I assigned a Product to a Tool within the Recipe form (Tool subforms were nested inside he Recipe form) everything has been saved correctly, but when I then edited the Recipe, the "Product" selectboxes on "Tool" subforms were empty (i.e. no selected value).

I created a custom field `ProductField < Administrate::Field::BelongsTo`, changed the `_form` partial so that the select looks like what @Billiam proposed and used this field in the ToolDashboard instead of Field::BelongsTo.

Now it works fine.

**Comment by NedelescuVlad** [comment]
I'm having the same issue for a HasMany relation that is 2 levels of nesting in, and couldn't get the solution @Billiam provided to work, as it is only for BelongsTo relations. Do you guys have any clue how I might go about this?
